spock-unitils/src/main/java/org/spockframework/unitils/UnitilsExtension.java

```diff
@@ -14,14 +14,14 @@
 
 package org.spockframework.unitils;
 
-import org.spockframework.runtime.extension.IAnnotationDrivenExtension;
+import org.spockframework.runtime.extension.IStatelessAnnotationDrivenExtension;
 import org.spockframework.runtime.model.*;
 import org.spockframework.util.NotThreadSafe;
 
 import spock.unitils.UnitilsSupport;
 
 @NotThreadSafe
-public class UnitilsExtension implements IAnnotationDrivenExtension<UnitilsSupport> {
+public class UnitilsExtension implements IStatelessAnnotationDrivenExtension<UnitilsSupport> {
```

Comment:
The `NotThreadSafe` annotation and the `IStatelessAnnotationDrivenExtension` makes me wonder.

But I do not see, why the class would not be `ThreadSafe`.

Reply:
I think the source is that `Unitils` works with a static singleton `org.unitils.core.Unitils#getInstance`